package muduo.net;
option java_package = "com.chenshuo.muduo.protorpc";
option java_outer_classname = "RpcProto";

enum MessageType
{
  REQUEST = 1;
  RESPONSE = 2;
  ERROR = 3;
}

enum ErrorCode
{
  NO_ERROR = 0;
  WRONG_PROTO = 1;
  NO_SERVICE = 2;
  NO_METHOD = 3;
  INVALID_REQUEST = 4;
  INVALID_RESPONSE = 5;
}

message RpcMessage
{
  required MessageType type = 1;
  required fixed64 id = 2;

  optional string service = 3;
  optional string method = 4;
  optional bytes request = 5;

  optional bytes response = 6;

  optional ErrorCode error = 7;
}
